controllers/chat/sendSignedMessage.js

```diff
@@ -101,18 +104,16 @@ const sendSignedMesage = async (req, res) => {
       if (isBlocked) {
         return res.status(403).json(responseError('This user does not want to receive messages'));
       }
-      // if (blockedIds.length > 0) {
-      //   for (const member of better_channel_member) {
-      //     if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
-      //       return res
-      //         .status(403)
-      //         .json(responseError('This user does not want to receive messages'));
-      //     }
-      //   }
-      // }
+
+      channelMember = better_channel_member.filter(
+        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+      );
+
+      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This logic filters out blocked users from the `better_channel_member` array and finds the sender's information. However, there's no null or undefined check for `better_channel_member`. Ensure that `better_channel_member` is not null or undefined before using filter or find on it.
```diff
+      if (better_channel_member) {
+        channelMember = better_channel_member.filter(
+          (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+        );
+
+        senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
+      }
-      channelMember = better_channel_member.filter(
-        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
-      );
-
-      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->